libraries/plugins/rc/rc_plugin.cpp

```diff
@@ -440,11 +430,335 @@ void rc_plugin_impl::on_first_block()
+      // transferring / delegating VESTS that haven't regenerated voting power.
+      //
+      // TODO:  Issue number
+      //
+      static_assert( STEEM_RC_REGEN_TIME <= STEEM_VOTING_MANA_REGENERATION_SECONDS, "RC regen time must be smaller than vote regen time" );
+
+      // ilog( "regenerate(${a})", ("a", account.name) );
+
+      if( account.vesting_shares != rc_account.last_vesting_shares )
+      {
+         if( !_skip.skip_reject_unknown_delta_vests )
+         {
+            STEEM_ASSERT( false, plugin_exception,
+               "Account ${a} VESTS changed from ${old} to ${new} without triggering an op, noticed on block ${b}",
+               ("a", account.name)("old", rc_account.last_vesting_shares)("new", account.vesting_shares)("b", _db.head_block_num()) );
+         }
+         else
+         {
+            elog( "Account ${a} VESTS changed from ${old} to ${new} without triggering an op, noticed on block ${b}",
+               ("a", account.name)("old", rc_account.last_vesting_shares)("new", account.vesting_shares)("b", _db.head_block_num()) );
+         }
+      }
+
+      manabar_params mbparams;
+      mbparams.max_mana = get_maximum_rc( account, rc_account );
+      mbparams.regen_time = STEEM_RC_REGEN_TIME;
+
+      _db.modify( rc_account, [&]( rc_account_object& rca )
+      {
+         rca.rc_manabar.regenerate_mana( mbparams, _current_time );
+      } );
+
+      _shared_state.regen_accounts.push_back( account.name );
+   }
+
+   template< bool account_may_not_exist = false >
+   void regenerate( const account_name_type& name )const
+   {
+      const account_object* account = _db.find< account_object, by_name >( name );
+      if( account_may_not_exist )
+      {
+         if( account == nullptr )
+            return;
+      }
+      else
+      {
+         FC_ASSERT( account != nullptr, "Unexpectedly, account ${a} does not exist", ("a", name) );
+      }
+
+      const rc_account_object* rc_account = _db.find< rc_account_object, by_name >( name );
+      FC_ASSERT( rc_account != nullptr, "Unexpectedly, rc_account ${a} does not exist", ("a", name) );
+
+      regenerate( *account, *rc_account );
+   }
+
+   void operator()( const transfer_to_vesting_operation& op )const
+   {
+      account_name_type target = op.to.size() ? op.to : op.from;
+      regenerate( target );
+   }
+
+   void operator()( const withdraw_vesting_operation& op )const
+   {
+      regenerate( op.account );
+   }
+
+   void operator()( const set_withdraw_vesting_route_operation& op )const
+   {
+      regenerate( op.from_account );
+   }
+
+   void operator()( const delegate_vesting_shares_operation& op )const
+   {
+      regenerate( op.delegator );
+      regenerate( op.delegatee );
+   }
+
+   void operator()( const author_reward_operation& op )const
+   {
+      regenerate( op.author );
+   }
+
+   void operator()( const curation_reward_operation& op )const
+   {
+      regenerate( op.curator );
+   }
+
+   // Is this one actually necessary?
```

Comment:
Probably not. But an account that is active since before the reward balance might have a different of a few satoshi.